src/bounding_hierarchy.rs

```diff
@@ -245,6 +247,85 @@ pub trait BoundingHierarchy<T: BHValue, const D: usize> {
         shapes: &'a [Shape],
     ) -> Vec<&'a Shape>;
 
+    /// Traverses the [`BoundingHierarchy`].
+    /// Returns the shape closest to the query point.
```

Comment:
A subjective concern: would the name `nearest_to` be better than `nearest_from`? Your documentation uses "to" here ;)